src/realm/exceptions.hpp

```diff
@@ -22,36 +22,52 @@
 #include <stdexcept>
 
 #include <realm/util/features.h>
+#include <realm/util/backtrace.hpp>
 
 namespace realm {
 
+template <class Base = std::runtime_error>
+class ExceptionWithBacktrace : public Base {
+public:
+    template <class... Args>
+    inline ExceptionWithBacktrace(Args&&... args)
+        : Base(std::forward<Args>(args)...)
+        , m_backtrace(util::Backtrace::capture())
+    {}
+
+    const util::Backtrace& backtrace() const noexcept { return m_backtrace; }
+
+private:
+    util::Backtrace m_backtrace;
+};
+
 /// Thrown by various functions to indicate that a specified table does not
 /// exist.
-class NoSuchTable : public std::exception {
+class NoSuchTable : public ExceptionWithBacktrace<std::exception> {
```

Comment:
This exception (`NoSuchTable`) is designed (by myself) an API exception, not a logic error. The same goes for the rest of the exceptions in this file except `LogicError` itself. I believe it is, at least conceptually, problematic to burden API exceptions with this scheme. We could do it in debug mode, though.